.github/workflows/commit.yml

```diff
@@ -0,0 +1,15 @@
+name: Lint commit messages
+
+on: 
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: wagoid/commitlint-github-action@v5
```

Comment:
Why use an action instead of running `npm ci && npm run commit:lint`? 🤔 This creates a difference between the local and CI workflows.

Reply:
Yep, I tried quickly but it [did not work as expected](https://github.com/Ndpnt/engine/actions/runs/8186461328/job/22384893953) and as this feature was not planned I did not take time to investigate.

Reply:
I believe this happens because, by default, the `checkout` action checks out only the current commit, for performance reasons :) so the `main` reference does not exist.

I can't try that right now as I'm on my phone, but if I recall correctly there is an option to pass to `checkout` (maybe `depth: 0`?) to check out the whole tree.

I believe it is rather important to unify CI and local tests and avoid an Action dependency if we can :)